app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt

```diff
@@ -65,12 +69,42 @@ class ProfileListAdapter(
             binding.TxtImg.setText(item.name[0].toString())
             binding.TxtImg.avatarBackgroundColor = bgColors[item.colorIndex]
             binding.timeStamp.text = item.timeInstance
+            binding.pauseSwitch.isChecked = item.pauseSwitch
 
             binding.profileCard.setOnClickListener {
                 val args = Bundle()
                 args.putParcelable("Profile", item)
                 val dialog = DetailsFragment.newInstance(args)
-                dialog.show((parentView.context as FragmentActivity).supportFragmentManager, "DialogFragment")
+                dialog.show(
+                    (parentView.context as FragmentActivity).supportFragmentManager,
+                    "DialogFragment"
+                )
+            }
+            binding.pauseSwitch.setOnClickListener {
+                if (!binding.pauseSwitch.isChecked) {
+                    item.pauseSwitch = false
+                    WorkManager.getInstance(profileViewModel.getApplication())
+                        .cancelAllWorkByTag(item.profileId.toString())
+                    if (StoreSession.readLong(AppConstants.ACTIVE_PROFILE_ID) == item.profileId)
+                        audioManager.ringerMode = StoreSession.readInt(AppConstants.RINGTONE_MODE)
+                } else {
+                    item.pauseSwitch = true
+                    val current: Calendar = Calendar.getInstance()
+                    if (StoreSession.readLong(AppConstants.ACTIVE_PROFILE_ID) == item.profileId) {
+                        if (current.get(Calendar.MINUTE) == 59) {
```

Comment:
@coder2699 Can you tell me what this if block is doing?

Reply:
@rob729 yeah....so basically firstly I check that if the user is using pause/resume feature them is that a currently active profile or not.

If profile is not currently active, then on rescheduling, the starting time remains same.

But if it is a currently active profile then this "if block" comes into the picture.
 When a profile is active and in the middle you stop it, then obviously on resuming, the starting time for rescheduling couldn't be the same (and needs to be equal to the current time). For that purpose I used this if block.

Reply:
@coder2699 Assume a case where a profile has start time: 8:50 PM and end time: 8:53 PM.
At 8:50 PM the profile got active and after it got active I paused the profile. Now at 8:52 PM I activate the profile then what will happen at 8:53 PM according to the code you have written?

Reply:
@rob729 According to the case that you have given, at 8:53 PM, app will get restored to its original ringer mode, since 8:53PM is the end time.